modules/porous_flow/include/kernels/PorousFlowHeatMassTransfer.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Kernel.h"
+
+// Forward Declaration
+class PorousFlowHeatMassTransfer;
+
+template <>
+InputParameters validParams<PorousFlowHeatMassTransfer>();
+
+/**
+ * Calculate heat or mass transfer from a coupled variable to u.
+ * This can be used to transfer from a variable specified in a highly conductive domain to a low
+ * conductive domain. With this kernel the retardation effect describted in the porous flow
+ * documentation can be overcome.
```

Comment:
Yes, I am planning on doing this. I create an example where this can be seen and will write something to be added to the documentation. I will create an issue and PR once I am done.